sites/docs/src/content/docs/tutorials/tests_and_test_data/components/04_testing_subworkflows.md

```diff
@@ -0,0 +1,323 @@
+                ])
+                input[1] = Channel.of([
+                            [ id:'test' ],
+                            file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
+                ])
+                // Use the output from the setup block
+                input[2] = BWA_INDEX.out.index
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert workflow.success },
+                { assert snapshot(
+                    workflow.out.bam.collect { meta, bamfile -> bam(bamfile).getReadsMD5() },
+                    workflow.out.bai.collect { meta, bai -> file(bai).name },
+                    workflow.out.stats.collect { meta, stats -> file(stats).name },
+                    workflow.out.flagstat.collect { meta, flagstat -> file(flagstat).name },
+                    workflow.out.versions
+                    ).match() }
+            )
+        }
+    }
+}
+```
+
+## Testing parameter variations
+
+Test different parameter combinations that affect subworkflow behavior:
+
+### Creating parameter-specific configuration
+
+Create `tests/nextflow.config` for subworkflow-specific process configuration:
+
+```groovy
+params {
+    aligner            = "bismark"
+    cytosine_report    = false
+    skip_deduplication = false
+}
+
+process {
+    withName: 'BISMARK_ALIGN' {
+        ext.args = { params.aligner == 'bismark_hisat' ? ' --hisat2' : ' --bowtie2' }
+    }
+
+    withName: 'SAMTOOLS_SORT' {
+        ext.prefix = { "${meta.id}.sorted" }
+    }
+}
+```
+
+### Overriding parameters with the `params` block
+
+In addition to a `nextflow.config` file, `nf-test` provides a `params` block within the `when` block to override Nextflow's input `params` for a specific test. This is useful for testing different parameter combinations without creating multiple config files.
+
+You can set parameters manually, including nested parameters:
+
+```groovy
+when {
+    params {
+        outdir = "output"
+        output {
+          dir = "output"
+        }
+    }
+    workflow {
+        """
+        // workflow inputs
+        """
+    }
+}
+```
+
+#### Loading parameters from a file
+
+You can also load parameters from a JSON or YAML file. This is useful for managing complex parameter sets.
+
+```groovy
+when {
+    params {
+        load("$baseDir/tests/params.json")
+    }
+    workflow {
+        // ...
+    }
+}
+```
+
+It is also possible to combine both techniques, allowing you to load a base set of parameters from a file and then override specific ones for a particular test case.
+
+```groovy
+when {
+    params {
+        load("$baseDir/tests/params.json")
+        outputDir = "new/output/path"
+    }
+    workflow {
+        // ...
+    }
+}
+```
+
+## Testing output channels comprehensively
+
+### BAM file testing with MD5 checksums
```

Comment:
Why are you singling out BAMs here? Also this is using a plugin not mentioned her yet.

I think these two sections make more sense for the assertion chaper.